### DISCUSSION TITLE
https://blend2d.com 2D library instead of SDL2

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [maximecb/uvm#6] [Ideas]
Adding 2D/framebuffer graphics right from the start is the strategically right decision to attract people.

Maybe you want to take a look at https://blend2d.com, which is a superior 2D library. But it might not fulfill your *needs to be portable to everything* requirement. However, the frame buffer component is central and should be chosen wisely.

Of course, keeping the integration and coupling as loose as possible to keep flexibility should be the goal.

**Comment by maximecb** [comment]
I chose SDL because it's very common and it also has bindings for audio. You can see the primitives for the APIs exposed here: https://github.com/maximecb/uvm/blob/main/doc/syscalls.md#window

I've tried to design those APIs in a way that they don't "leak" SDL-specific details, which makes it possible to yank out SDL later if needed, and replace it with X11/ALSA or something else.
